server/src/main/java/org/elasticsearch/cluster/metadata/AutoExpandReplicas.java

```diff
@@ -110,20 +109,21 @@ public OptionalInt getDesiredNumberOfReplicas(IndexMetadata indexMetadata, Routi
                 numMatchingDataNodes++;
             }
         }
+        return calculateDesiredNumberOfReplicas(numMatchingDataNodes);
+    }
 
+    // Package private only for testing purposes!
+    int calculateDesiredNumberOfReplicas(int numMatchingDataNodes) {
         final int min = minReplicas();
         final int max = getMaxReplicas(numMatchingDataNodes);
```

Comment:
The `Math.min` done in `getMaxReplicas` seems redundant now, I think we should remove it, it is somewhat confusing now.

Reply:
That's a very good point! It also simplifies the code and the parsing test. Done in https://github.com/elastic/elasticsearch/pull/87505/commits/ec7d7f97bc039f46d8c508943783b44ea78edae2.